lib/manageiq/rpm_build/rpm_repo.rb

```diff
@@ -0,0 +1,77 @@
+module ManageIQ
+  module RPMBuild
+    class RpmRepo
+      include S3Common
+
+      def update
+        require 'tempfile'
+        Dir.mktmpdir do |tmpdir|
+          work_dir = Pathname.new(tmpdir)
+
+          # Build directory structure /release/{master,jansa}/el8/{noarch,src,x86_64}
+          require 'fileutils'
+          directories = []
+          OPTIONS.rpm_repository.content.each do |branch, values|
+            values[:targets].each do |target|
+              OPTIONS.rpm_repository.arches.each do |arch|
+                directories << work_dir.join("release", branch.to_s, target, arch)
+              end
+            end
+          end
+          directories.each { |dir| FileUtils.mkdir_p(dir) }
+
+          # Fetch the configured RPMs
+          puts "Downloading required RPMs..."
+          require 'rest-client'
+          OPTIONS.rpm_repository.content.each do |branch, values|
+            values[:rpms]&.each do |rpm, versions|
+              remote_url = File.join("https://#{OPTIONS.rpm_repository.s3_api.bucket}.#{OPTIONS.rpm_repository.s3_api.endpoint}", "?prefix=builds", rpm.to_s)
+              files      = RestClient.get(remote_url).body.scan(/<Key>([^<>]*)<\/Key>/).flatten
```

Comment:
As much as we want to avoid XML, I think it's safer to use Nokogiri, and then use some XPath expressions to find what we need

Reply:
Also, is this XML structure an S3 thing or is this our DO version?  If S3, I'd expect some method in aws-sdk-s3 to handle it.